### PR TITLE
docs(eslint-plugin): add Deprecating Formatting Rules blog post

### LINKED ISSUE #8115: Docs: Add quick blog post about deprecating formatting rules
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Following #8072 -> #8073 and preceding #8074: I'm thinking it'd be good to have a brief blog post mentioning that we're following ESLint in deprecating formatting rules & deferring to [eslint-stylistic](https://eslint.style) for their maintenance. Any kind of mass deprecation->deletion is better done with lots of messaging beforehand.

### Affected URL(s)

https://typescript-eslint.io/blog/deprecating-formatting-rules, maybe?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8115
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Summarizes why we're doing this and links to the ESLint blog post + ESLint Stylistic docs as well.

Also includes a fix to the rule replacements: I'd forgotten `ts/` exists as a separate namespace from `default/`.

### CODE REVIEW

File: packages/website/blog/2023-12-25-deprecating-formatting-rules.md
Comment (JoshuaKGoldberg) [inline packages/website/blog/2023-12-25-deprecating-formatting-rules.md:1]: cc @antfu @typescript-eslint/triage-team - I wish I'd thought to write this post alongside #8072/#8073. Ah well. I've set a reminder to merge this PR in on Monday before our scheduled weekly release.

**Review by Josh-Cena — APPROVED** [review]
Content looks great; not sure if the code change should be a separate PR but no strong feelings

File: packages/eslint-plugin/src/rules/lines-between-class-members.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default createRule<Options, MessageIds>({
   name: 'lines-between-class-members',
   meta: {
     deprecated: true,
-    replacedBy: ['@stylistic/line-between-class-members'],
+    replacedBy: ['@stylistic/ts/line-between-class-members'],
```
Comment (auvred) [inline packages/eslint-plugin/src/rules/lines-between-class-members.ts:37]:
```suggestion
    replacedBy: ['@stylistic/ts/lines-between-class-members'],
```

**Review by auvred — CHANGES_REQUESTED** [review]
Nice post!

I just found a few typos

---

By the way, maybe this has already been discussed somewhere, but I missed it:
What about [`no-extra-semi`](https://typescript-eslint.io/rules/no-extra-semi) rule? It's classified as `suggestion` rather than `layout`, but [it's already a part](https://github.com/eslint-stylistic/eslint-stylistic/blob/de02f0008a8293c44f7d77270a927cdfa936ed46/packages/eslint-plugin-ts/rules/index.ts#L41) of `eslint-stylistic`

Shouldn't we deprecate it as well? Because otherwise it will be duplicated in both `ts-eslint` and `eslint-stylistic`

**Comment by Josh-Cena** [comment]
I asked a while ago privately; I think Brad is in favor of removing it too :)

File: packages/website/blog/2023-12-25-deprecating-formatting-rules.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+| [`@typescript-eslint/indent`](/rules/indent)                                                  | [`@stylistic/indent`](https://eslint.style/rules/ts/indent)                                           |
+| [`@typescript-eslint/key-spacing`](/rules/key-spacing)                                        | [`@stylistic/key-spacing`](https://eslint.style/rules/ts/key-spacing)                                 |
+| [`@typescript-eslint/keyword-spacing`](/rules/keyword-spacing)                                | [`@stylistic/keyword-spacing`](https://eslint.style/rules/ts/keyword-spacing)                         |
+| [`@typescript-eslint/lines-around-comment`](/rules/lines-around-comment)                      | [`@stylistic/lines-around-comment`](https://eslint.style/rules/ts/lines-around-comment)               |
+| [`@typescript-eslint/lines-between-class-members`](/rules/lines-between-class-members)        | [`@stylistic/line-between-class-members`](https://eslint.style/rules/ts/between-class-members)        |
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/website/blog/2023-12-25-deprecating-formatting-rules.md:54]: Thanks :)

**Comment by JoshuaKGoldberg** [comment]
> if the code change should be a separate PR

Heh yeah in theory maybe. But it's all going to ship at the same time.

**Comment by antfu** [comment]
Thank you! And Merry Christmas to you all!

**Pull request merged**